### ISSUE TITLE
Spike for Indonesia e-commerce content operations 

### ISSUE THREAD

**Issue by joechengla** [mozilla-tw/FirefoxLite#3229]
As Mozilla, we need to figure out how to operate E-Commerce Content portal
1. When and How to export campaign assets from partner backend
2. How to decide what campaigns to surface 
3. Where to store the campaign assets
4. What format to store the campaign assets
5. When/How to take campaigns down

**Comment by chianger** [comment]
This is prerequisite of #1999

**Comment by chianger** [comment]
taking this into sprint P but to be clear content portal is still higher priority whenever there's resource conflict.

**Comment by cnevinc** [comment]
Here's the estimation
https://docs.google.com/spreadsheets/d/1FISkSKCrMGEYJe2ok3zFuNjkTzdNKTDmUGSAltFS1-Y/edit#gid=0

tldr:
- 5.38 sprint for 1 developer
- 3.13 sprint for 2 developers

**Comment by cnevinc** [comment]
> As Mozilla, we need to figure out how to operate E-Commerce Content portal
> 
> 1. When and How to export campaign assets from partner backend
human currated content and make it a configuration file in our/Firebase backend

> 2. How to decide what campaigns to surface
Marketing team will decide

> 3. Where to store the campaign assets
Firebase or our own backend

> 4. What format to store the campaign assets
TBD

> 5. When/How to take campaigns down
Make it an engineering feature

**Comment by cnevinc** [comment]
Here's a sample data from the backend
we should consider using a second layer json file to prevent spending too much time on RemoteConfig fetching..

```
[
  {
    "_id": "5c8f5b51d2e170e5fa8d2310",
    "isActive": false,
    "isHighlight": false,
    "title": "Beli Tiket Kereta Lebih Irit, Mudik Jadi Lebih Asik",
    "promo_url": "http://bukalapak.go2cloud.org/aff_c?offer_id=15&aff_id=4287&url=https%3A%2F%2Fwww.bukalapak.com%2Fpromo-detail%2Fbeli-tiket-kereta-lebih-irit-mudik-jadi-lebih-asik%3Fho_offer_id%3D{offer_id}%26ho_trx_id%3D{transaction_id}%26affiliate_id%3D{affiliate_id}%26utm_source%3Dhasoffers%26utm_medium%3Daffiliate%26utm_campaign%3D{offer_id}%26ref%3D{referer}",
    "start": "2016-03-08T02:08:38 -08:00",
    "end": "2015-06-27T12:33:16 -08:00",
    "code": "BUSPERGI",
    "thumbnail": "http://placehold.it/32x32",
    "banner": "http://placehold.it/600x320"
  },
  {
    "_id": "5c8f5b51f0dc2bac78d20e4d",
    "isActive": false,
    "isHighlight": true,
    "title": "Beli Tiket Kereta Lebih Irit, Mudik Jadi Lebih Asik",
    "promo_url": "http://bukalapak.go2cloud.org/aff_c?offer_id=15&aff_id=4287&url=https%3A%2F%2Fwww.bukalapak.com%2Fpromo-detail%2Fbeli-tiket-kereta-lebih-irit-mudik-jadi-lebih-asik%3Fho_offer_id%3D{offer_id}%26ho_trx_id%3D{transaction_id}%26affiliate_id%3D{affiliate_id}%26utm_source%3Dhasoffers%26utm_medium%3Daffiliate%26utm_campaign%3D{offer_id}%26ref%3D{referer}",
    "start": "2014-10-01T07:43:33 -08:00",
    "end": "2017-05-15T02:40:39 -08:00",
    "code": "BUSPERGI",
    "thumbnail": "http://placehold.it/32x32",
    "banner": "http://placehold.it/600x320"
  }
]
```

**Comment by cnevinc** [comment]
I've put the sample in Firebase Hosting
https://rocket-dev01.firebaseapp.com/shopping.json

**Comment by chianger** [comment]
efforts estimation and draft json format were given

**Comment by cnevinc** [comment]
updated the sample data
[
   {
      "source":"bukalapak",
      "name":"PROMO Oishi Sponge Crunch Rasa Coklat Ukuran 120 Gram",
      "img":"https://s4.bukalapak.com/img/9946531644/w-300/PROMO_Oishi_Sponge_Crunch_Rasa_Stowberi_Ukuran_120_Gram.jpg.webp",
      "url":"http://bukalapak.go2cloud.org/aff_c?offer_id=15&aff_id=4287&aff_sub=coupon&url=https%3A%2F%2Fwww.bukalapak.com%2Fp%2Ffood%2Fcemilan-snack%2Fy1qe6a-jual-promo-oishi-sponge-crunch-rasa-coklat-ukuran-120-gram%3Ffrom%3Domnisearch%26search_source%3Domnisearch_history%26source%3Dnavbar%26ho_offer_id%3D{offer_id}%26ho_trx_id%3D{transaction_id}%26affiliate_id%3D{affiliate_id}%26utm_source%3Dhasoffers%26utm_medium%3Daffiliate%26utm_campaign%3D{offer_id}%26ref%3D{referer}",
      "start":1553011200000,
      "end":1558281600000,
      "active":true
   },
   {
      "source":"bukalapak",
      "name":"Oppo F9 Dual Sim Ram 4 Rom 64gb",
      "img":"https://s4.bukalapak.com/img/4723640188/w-300/data.png.webp",
      "url":"http://bukalapak.go2cloud.org/aff_c?offer_id=15&aff_id=4287&aff_sub=coupon&url=https%3A%2F%2Fwww.bukalapak.com%2Fp%2Fhandphone%2Fhp-smartphone%2F1iiyj2k-jual-oppo-f9-dual-sim-ram-4-rom-64gb%3Ffrom%3Domnisearch%26search_source%3Domnisearch_history%26source%3Dnavbar%26ho_offer_id%3D{offer_id}%26ho_trx_id%3D{transaction_id}%26affiliate_id%3D{affiliate_id}%26utm_source%3Dhasoffers%26utm_medium%3Daffiliate%26utm_campaign%3D{offer_id}%26ref%3D{referer}",
      "start":1553011200000,
      "end":1558281600000,
      "active":true
   },
   {
      "source":"bukalapak",
      "name":"VIVO V15 RED",
      "img":"https://s4.bukalapak.com/img/4430537728/w-300/data.png.webp",
      "url":"http://bukalapak.go2cloud.org/aff_c?offer_id=15&aff_id=4287&aff_sub=coupon&url=https%3A%2F%2Fwww.bukalapak.com%2Fp%2Fhandphone%2Fhp-smartphone%2F1fu21c1-jual-vivo-v15-red%3Ffrom%3Domnisearch%26search_source%3Domnisearch_history%26source%3Dnavbar%26ho_offer_id%3D{offer_id}%26ho_trx_id%3D{transaction_id}%26affiliate_id%3D{affiliate_id}%26utm_source%3Dhasoffers%26utm_medium%3Daffiliate%26utm_campaign%3D{offer_id}%26aff_sub%3D{aff_sub}%26ref%3D{referer}",
      "start":1553011200000,
      "end":1558281600000,
      "active":true
   }
]
